code/datums/weather/weather_types/solar_flare.dm

```diff
@@ -39,13 +39,12 @@
 	SSsun.solar_gen_rate = initial(SSsun.solar_gen_rate) * 40
 
 /datum/weather/solar_flare/can_weather_act(mob/living/L)
-	. = ..()
-	if(.) //If true the mob is already affected, no need to keep processing
-		return TRUE
 	if(isanimal(L)) //while this might break immersion, I don't want to spam the server with calling this on simplemobs
 		return FALSE
 	if(isdrone(L)) //same with poor maint drones who just wanna have fun
 		return FALSE
+	if(!..(L)) //If true the mob is already affected, no need to keep processing
```

Comment:
Thank you for implementing the changes to fix this, it might be easier to put this back as it was but inverted, for readability sake? I.E.
```
. = ..()
	if(!.) // If false the mob is not currently a valid target, no need to keep processing
		return FALSE

Reply:
It has been done